src/Features/Core/Portable/SolutionCrawler/WorkCoordinator.IncrementalAnalyzerProcessor.cs

```diff
@@ -100,11 +105,73 @@ public void Enqueue(WorkItem item)
                 {
                     Contract.ThrowIfNull(item.DocumentId);
 
-                    _highPriorityProcessor.Enqueue(item);
-                    _normalPriorityProcessor.Enqueue(item);
-                    _lowPriorityProcessor.Enqueue(item);
+                    var options = _registration.Workspace.Options;
+                    var analysisScope = SolutionCrawlerOptions.GetBackgroundAnalysisScope(options);
+
+                    if (ShouldEnqueueForAllQueues(item, analysisScope))
+                    {
+                        _highPriorityProcessor.Enqueue(item);
+                        _normalPriorityProcessor.Enqueue(item);
+                        _lowPriorityProcessor.Enqueue(item);
+                    }
+                    else
+                    {
+                        if (ShouldEnqueueReducedWorkItem(item, _highPriorityProcessor.Analyzers, options, analysisScope, out var reducedWorkItem))
+                        {
+                            _highPriorityProcessor.Enqueue(reducedWorkItem.Value);
+                        }
+
+                        if (ShouldEnqueueReducedWorkItem(item, _normalPriorityProcessor.Analyzers, options, analysisScope, out reducedWorkItem))
+                        {
+                            _normalPriorityProcessor.Enqueue(reducedWorkItem.Value);
+                        }
+
+                        if (ShouldEnqueueReducedWorkItem(item, _lowPriorityProcessor.Analyzers, options, analysisScope, out reducedWorkItem))
+                        {
+                            _lowPriorityProcessor.Enqueue(reducedWorkItem.Value);
+                        }
+                    }
 
                     ReportPendingWorkItemCount();
+
+                    return;
+
+                    bool ShouldEnqueueForAllQueues(WorkItem item, BackgroundAnalysisScope analysisScope)
+                    {
+                        var reasons = item.InvocationReasons;
+                        if (analysisScope == BackgroundAnalysisScope.ActiveFile &&
+                            !reasons.Contains(PredefinedInvocationReasons.DocumentClosed) &&
+                            !reasons.Contains(PredefinedInvocationReasons.DocumentRemoved))
+                        {
+                            // Only process active/closed/removed documents for active file analysis.
+                            return item.DocumentId == _documentTracker.TryGetActiveDocument();
+                        }
+
+                        return true;
+                    }
+                }
+
+                private static bool ShouldEnqueueReducedWorkItem(
```

Comment:
ReducedWorkItem doesn't really mean anything to me.  It seems like it's more TryGetItemWithOverriddenAnalysisScope